GCP/deploy.sh

```diff
@@ -84,31 +106,31 @@ fi
 echo "----- Deploying worker nodes... -----"
 for worker in $CLUSTER_NAME-worker1 $CLUSTER_NAME-worker2 $CLUSTER_NAME-worker3
 do
-  host=`gcloud compute instances describe --project=$PROJECT --zone=$ZONE $worker --format='get(networkInterfaces[0].accessConfigs[0].natIP)'`
+  host=$(gcloud compute instances describe --project="$PROJECT" --zone="$ZONE" "$worker" --format='get(networkInterfaces[0].accessConfigs[0].natIP)')
 
-  ssh -q -o "StrictHostKeyChecking=no" $user@$host "echo ${token} > /tmp/token.txt && curl -sfL https://get.k3s.io | sudo sh -s - agent --server https://${master_private}:6443 --token-file /tmp/token.txt --node-external-ip ${host}" &>/dev/null  &
+  ssh -i "$privkeypath" -q -o "StrictHostKeyChecking=no" $user@"$host" "echo ${token} > /tmp/token.txt && curl -sfL https://get.k3s.io | sudo sh -s - agent --server https://${master_private}:6443 --token-file /tmp/token.txt --node-external-ip ${host}" &>/dev/null  &
 done
 
 echo "----- Deployment finished... waiting for all the nodes to become k3s ready... -----"
 if [ "$load_kube_config" = "true" ]
 then
-  nodes_check=`kubectl get nodes | grep Ready | wc -l | tr -d ' '`
+  nodes_check=$(kubectl get nodes | grep -c Ready | tr -d ' ')
```

Comment:
Did you check this on both MacOS and Linux?

Reply:
No, only Linux. Just checked and it's not the same `grep` in both case: GNU grep vs BSD grep. It seems the `-c` argument is available on BSD grep as well tho, cf https://man.bsd.lv/grep.1 , so it should work on macOS.